lib/maze/schemas/validator.rb

```diff
@@ -1,55 +1,46 @@
 # frozen_string_literal: true
 
+require_relative 'config_validator'
+require_relative 'error_validator'
+require_relative 'trace_validator'
+
 module Maze
   module Schemas
 
     # A general entry point for running validation with schemas and other validation methods
     class Validator
 
       class << self
-
-        # Tests that payloads for a specific path have passed any schema checks implemented on receipt
-        # Throws an AssertionFailedError with a list of issues on failure
-        #
-        # @param list [Array] An array of received requests
-        # @param list_name [String] The name of the payload list for received requests
-        def verify_against_schema(list, list_name)
-          request_schema_results = list.all.map { |request| request[:schema_errors] }
-          passed = true
-          request_schema_results.each.with_index(1) do |schema_errors, index|
-            next if schema_errors.nil?
-            if schema_errors.size > 0
-              passed = false
-              $stdout.puts "\n"
-              $stdout.puts "\e[31m--- #{list_name} #{index} failed validation:\e[0m"
-              schema_errors.each do |error|
-                $stdout.puts "\e[31m#{JSONSchemer::Errors.pretty(error)}\e[0m"
-              end
-              $stdout.puts "\n"
-            end
-          end
-
-          unless passed
-            raise Test::Unit::AssertionFailedError.new 'The received payloads did not match the endpoint schema.  A full list of the errors can be found above'
-          end
-        end
-
         # Tests that payloads for a specific path pass any additional validation checks
         # Throws an AssertionFailedError with a list of issues on failure
         #
         # @param list [Array] An array of received requests
         # @param list_name [String] The name of the payload list for received requests
         def validate_payload_elements(list, list_name)
-          validator_class = case list_name
-          when 'trace', 'traces'
-            Maze::Schemas::TraceValidator
+          # Test to see if a custom validator exists for the list
+          if Maze.config.custom_validators&.key?(list_name)
+            custom_validator = true
+            validator_class = Maze::Schemas::ConfigValidator
           else
-            nil
+            custom_validator = false
+            validator_class = case list_name
+            when 'trace', 'traces'
+              Maze::Schemas::TraceValidator
+            when 'error', 'errors'
+              Maze::Schemas::ErrorValidator
+            else
+              nil
+            end
```

Comment:
Yes, that's correct.  This was more of a concession to the ongoing discussion over tweaking validation in different circumstances.  I feel like this could be added to, to form three levels of operation:
- Run built-in validation by default
- If a validation block is configured, run that on top of built-in validation
- Built in validation can always be turned on and off, at a scenario, feature, or test-run level (but any configured validation will be run)